core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -502,6 +522,8 @@ private[spark] class BlockManagerInfo(
 
   def blocks: JHashMap[BlockId, BlockStatus] = _blocks
 
+  def cachedBlocks: mutable.HashSet[BlockId] = _cachedBlocks
```

Comment:
what I meant was change the return type to an immutable set; we can just return the underlying mutable set without cloning it. The point is just to discourage users of cached blocks from mutating it (they still can if they want, but they need to cast it first)